File: src/main/java/toby/jpa/service/IExcuseService.java
package toby.jpa.service;

import toby.jpa.dto.ExcuseDto;

import java.util.List;

public interface IExcuseService {

    List<ExcuseDto> listAllGuildExcuses(Long guildId);
    List<ExcuseDto> listApprovedGuildExcuses(Long guildId);
    List<ExcuseDto> listPendingGuildExcuses(Long guildId);
    ExcuseDto createNewExcuse(ExcuseDto excuseDto);
    ExcuseDto getExcuseById(Integer id);
    ExcuseDto updateExcuse(ExcuseDto excuseDto);
    void deleteExcuseByGuildId(Long guildId);
    void deleteExcuseById(Integer id);
    void clearCache();

}

File: database/src/main/kotlin/database/service/IExcuseService.kt
package database.service

import database.dto.ExcuseDto

interface IExcuseService {
    fun listAllGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun listApprovedGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun listPendingGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun createNewExcuse(excuseDto: ExcuseDto?): ExcuseDto?
    fun getExcuseById(id: Long?): ExcuseDto?
    fun updateExcuse(excuseDto: ExcuseDto): ExcuseDto?
    fun deleteExcuseByGuildId(guildId: Long?)
    fun deleteExcuseById(id: Long?)
    fun clearCache()
}
